packages/router/index.js

```diff
@@ -6,67 +6,99 @@ const { log } = require("@spaship/common/lib/logging/pino");
 const config = require("./config");
 const { difference } = require("lodash");
 
-let topLevelDirs = [];
+let topLevelDirsCache = [];
 
 async function updateDirCache() {
   // Load directory names into memory
   const freshDirs = await fsp.readdir(config.get("webroot"));
 
-  const added = difference(freshDirs, topLevelDirs);
-  const removed = difference(topLevelDirs, freshDirs);
+  const added = difference(freshDirs, topLevelDirsCache);
+  const removed = difference(topLevelDirsCache, freshDirs);
 
   // update cache
-  topLevelDirs = freshDirs;
+  topLevelDirsCache = freshDirs;
+
   // Sort them by name length from longest to shortest, for most specific match wins matching policy
-  topLevelDirs.sort((a, b) => b.length - a.length);
+  topLevelDirsCache.sort((a, b) => b.length - a.length);
 
   if (added.length || removed.length) {
     log.info({ dirCache: { added, removed } }, `detected changes in SPAship directory list`);
   }
 }
 
-const customRouter = function(req) {
+const customRouter = function (req) {
   let url = req.url.replace(/\/+/g, "/"); // Avoid duplicate slash issue
-  let routeUrl;
-  let match;
-  let spaPath;
-  let matchedFlatDir;
+  let routeUrl; // This is the final path that the router will route the request to
+  let regExMatch; // Regular expression match object for finding SPA paths in the url
```

Comment:
If a variable should not change, you should consider to replace `let` to `const`

Reply:
Agreed, all of these variables change though so they have to be `let`